doc/api/fs.md

```diff
@@ -332,6 +332,33 @@ If one or more `filehandle.read()` calls are made on a file handle and then a
 position till the end of the file. It doesn't always read from the beginning
 of the file.
 
+#### `filehandle.readJSON(options)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `options` {Object|string}
+  * `encoding` {string|null} **Default:** `null`
+  * `reviver` {Function} Reviver function passed to [`JSON.parse`][].
+  * `signal` {AbortSignal} allows aborting an in-progress readJSON
+* Returns: {Promise} Fulfills upon a successful read with the contents of the
+  file parsed with [`JSON.parse`][].
+
+Asynchronously reads the contents of a file and call [`JSON.parse`][] on it.
+
+The `encoding` and `signal` options are passed to the underlying
+[`fsPromises.readFile()`][] call and the `reviver` option is passed to
+[`JSON.parse`][].
+
+If `options` is a string, then it specifies the `encoding`.
+
+The {FileHandle} has to support reading.
+
+If one or more `filehandle.read()` calls are made on a file handle and then a
+`filehandle.readJSON()` call is made, the data will be read from the current
+position till the end of the file. It doesn't always read from the beginning
+of the file.
```

Comment:
What happens if `readJSON()` is called multiple times?

Reply:
Specifically, it would be fantastic if this could be used to read new-line delimited json (a single file with multiple individual json objects embedded)

Reply:
Do you think it should be the only way `filehandle.json()` works or an option added later?

Reply:
An option would work. It would change the way the file is read, of course, potentially making it buffer. You'd have to stop at new lines which could get complicated. For now, maybe just indicate clearly in the docs that the operation reads to end of the file

Reply:
Yeah, it would be really *cool* if we could make this just stop reading from the file whenever the JSON input is complete -- it's obviously trickier, but it's something that would make this method a lot more powerful

Reply:
I wonder if I should remove the `filehandle.readJSON` for now. I agree with you that it would be a lot more powerful but I don't know how I would implement it and I don't want it to block the rest of the PR.

Reply:
No I wouldn't remove it. It's still useful as is. I'd just look at making the partial read an option later in a separate pr.